indico/web/util.py

```diff
@@ -179,3 +181,32 @@ def get_request_info(hide_passwords=True):
 def url_for_index(_external=False, _anchor=None):
     from indico.web.flask.util import url_for
     return url_for('categories.display', _external=_external, _anchor=_anchor)
+
+
+def signed_url_for(user, blueprint, url_params=None, *args, **kwargs):
+    from indico.web.flask.util import url_for
+    _external = kwargs.pop('_external', False)
+    base_url = url_for(blueprint, *args, **(url_params or {}))
+    qs = url_encode(sorted(kwargs.items()))
+    url = '{}?{}'.format(base_url, qs) if qs else base_url
+    signer = Signer(user.signing_secret, salt='url-signing')
+    qs = url_encode(dict(kwargs, token=signer.get_signature(url)))
+    full_base_url = url_for(blueprint, *args, _external=_external, **(url_params or {}))
+    return '{}?{}'.format(full_base_url, qs) if qs else base_url
+
+
+def is_signed_url_valid(user, url):
+    parsed = url_parse(url)
+    params = url_decode(parsed.query)
+    if 'token' not in params:
+        return False
+    signature = params.pop('token')
```

Comment:
```python
try:
    signature = params.pop('token')
except KeyError:
    return False
```

-> no need for the check above anymore

(and especially now that the `[0]` is gone I think this is the cleanest, most readable and most pythonic option)